### PR TITLE
1377 prod CICD for IHE GW

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#1377

### Dependencies

- Upstream: 
   - https://github.com/metriport/metriport/pull/1590
   - https://github.com/metriport/metriport-internal/pull/1546
- Downstream: none

### Description

Add `prod` CICD for IHE GW.

### Testing

- Sandbox
  - [ ] nothing for IHE GW
- Production
  - [ ] changes on config result in pushing those to the server
  - [ ] changes on dockerfile re-redeploy the server

### Release Plan

- [ ] Manually deploy the IHE Stack to `prod`
- set GH vars
   - [ ] `ECR_REPO_URI_PRODUCTION`
   - [ ] `ECS_CLUSTER_PRODUCTION`
   - [ ] `IHE_INBOUND_ECS_SERVICE_PRODUCTION`
   - [ ] `IHE_OUTBOUND_ECS_SERVICE_PRODUCTION`
- [ ] merge this
- [ ] Update IHE GW config on `staging`

### CODE REVIEW

File: .github/workflows/deploy-production.yml [hunk 1]
```diff
@@ -142,13 +150,53 @@ jobs:
...
+    # TODO 1377 revert this
+    # needs: [api, infra-api-lambdas-sandbox, ihe-gw-config]
+    needs: [api-sandbox, infra-api-lambdas-sandbox, ihe-gw-server]
     # run even if one of the dependencies didn't
     # can't use ${{ ! failure() && success() }} because `success()` "Returns true when none of the previous steps have failed or been canceled."
     # can't use ${{ ! failure() && contains(needs.*.result, 'success') }} because if anything that came before succeeded, even if not a direct dependency, it will run
-    if: ${{ !failure() && (needs.api-sandbox.result == 'success' || needs.infra-api-lambdas-sandbox.result == 'success') }}
+    # TODO 1377 revert this
+    # if: ${{ !failure() && (needs.api-sandbox.result == 'success' || needs.infra-api-lambdas-sandbox.result == 'success' || needs.ihe-gw-config.result == 'success') }}
+    if: ${{ !failure() && (needs.api-sandbox.result == 'success' || needs.infra-api-lambdas-sandbox.result == 'success' || needs.ihe-gw-server.result == 'success') }}
```
Comment (Goncharo) [inline .github/workflows/deploy-production.yml:199]: this comment is the same as the current config?

File: .github/workflows/deploy-production.yml [hunk 2]
```diff
@@ -142,13 +150,53 @@ jobs:
...
+  #     IHE_GW_URL: ${{ secrets.IHE_GW_URL_PRODUCTION }}
+  #     IHE_GW_USER: ${{ secrets.IHE_GW_USER }}
+  #     IHE_GW_PASSWORD: ${{ secrets.IHE_GW_PASSWORD }}
+
   release:
     uses: ./.github/workflows/_release.yml
-    needs: [api-sandbox, infra-api-lambdas-sandbox]
+    # TODO 1377 revert this
+    # needs: [api, infra-api-lambdas-sandbox, ihe-gw-config]
+    needs: [api-sandbox, infra-api-lambdas-sandbox, ihe-gw-server]
```
Comment (Goncharo) [inline .github/workflows/deploy-production.yml:193]: why is this necessary?

File: .github/workflows/deploy-production.yml [hunk 3]
```diff
@@ -142,13 +150,53 @@ jobs:
...
+      IHE_GW_CONFIG_BUCKET_NAME: ${{ vars.IHE_GW_CONFIG_BUCKET_NAME }}
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      IHE_GW_KEYSTORE_STOREPASS: ${{ secrets.IHE_GW_KEYSTORE_STOREPASS }}
+      IHE_GW_KEYSTORE_KEYPASS: ${{ secrets.IHE_GW_KEYSTORE_KEYPASS }}
+
+  # ihe-gw-config:
```
Comment (Goncharo) [inline .github/workflows/deploy-production.yml:176]: assuming this will be added back after the reverts in the TODOs?

**Review by Goncharo — APPROVED** [review]
conditional approval based on answers to the Qs

File: .github/workflows/deploy-production.yml (hunk 3 above)
Comment (leite08) [reply .github/workflows/deploy-production.yml:176]: This is following the same pattern as staging, we still don't have CI/CD for the IHE GW config. Need to expose an endpoint on IHE API GW to GH Actions can call it to update the configs.

File: .github/workflows/deploy-production.yml (hunk 2 above)
Comment (leite08) [reply .github/workflows/deploy-production.yml:193]: So we only execute release after `ihe-gw-server` is finished - in addition to API and Infra.

File: .github/workflows/deploy-production.yml (hunk 1 above)
Comment (leite08) [reply .github/workflows/deploy-production.yml:199]: Yes

**Review by Orta21 — APPROVED** [review]
